web/scripts/retrieve_data.py

```diff
@@ -224,30 +270,6 @@ def clean_wiki_description(wiki_desc: str) -> str:
     return clean_wiki_desc
 
 
-def add_ranks(rows: list, metrics: list) -> None:
```

Comment:
Here's another point we need to discuss. Do you want to:

1.) Calculate rank etc in js
2.) Have me add a new object which gives rankings for all the columns we might include in list view, whether they're from `patents` or `articles` or a top-level metric like `tt1_jobs`
3.) Have me add the ranking metrics to `patents` and `articles` when applicable, otherwise add them at the top level
4.) Have me create a third object akin to `patents` and so on but just for top-level metrics, with a null value for `counts` (the per-year counts column), and put these metrics in there where applicable, otherwise in `patents` and `articles` <-- I kind of like this one best in theory, it seems clean, but it might necessitate a ton of changes on your end so happy to do something else

Reply:
(discussion with Brian: option 4)